js/DrupalImageEditor.es6.js

```diff
@@ -1,179 +1,148 @@
-        };
-        img.onerror = (error) => {
-          console.error('upload error', error);
-        };
-        img.src = this.container.querySelector('a').href;
-      }
-    });
-  },
 
-  refreshPreview: function() {
-    if (this.last_preview === this.preview_value) {
-      return;
-    }
-    this.last_preview = this.preview_value;
-    this.preview.innerHTML = '';
-    if (!this.preview_value) {
-      return;
-    }
-    let mime = this.preview_value.match(/^data:([^;,]+)[;,]/);
-    if (mime) {
-      mime = mime[1];
+    if (this.button) {
+      this.container.appendChild(this.button);
     }
-    else {
-      mime = 'unknown';
-    }
-    let file = this.urlfield.files[0];
-    this.preview.innerHTML = '<strong>Type:</strong> ' + mime + ', <strong>Size:</strong> ' + file.size + ' bytes';
-    if (mime.substr(0, 5) === "image") {
-      this.preview.innerHTML += '<br>';
+
+    window.requestAnimationFrame(() => {
+      this.refreshPreview();
+    })
+  },
+  afterInputReady: function () {
+    if (this.value) {
       let img = document.createElement('img');
       img.style.maxWidth = '100%';
       img.style.maxHeight = '100px';
-      img.src = this.preview_value;
-      this.preview.appendChild(img);
+      img.onload = (event) => {
+        this.preview.appendChild(img)
+      };
+      img.onerror = (error) => {
+        console.error('upload error', error, this)
+      };
+      img.src = this.container.querySelector('input').value;
     }
-    this.preview.innerHTML += '<br>';
-    let uploadButton = this.getButton('Upload', 'upload', 'Upload');
-    this.preview.appendChild(uploadButton);
-    uploadButton.addEventListener('click', (event) => {
-      event.preventDefault();
-      uploadButton.setAttribute("disabled", "disabled");
-      this.theme.removeInputError(this.uploader);
-      if (this.theme.getProgressBar) {
-        this.progressBar = this.theme.getProgressBar();
-        this.preview.appendChild(this.progressBar);
-      }
-      this.jsoneditor.options.upload(this.path, file, {
-        success: (url) => {
-          this.setValue(url);
-          if (this.parent) {
-            this.parent.onChildEditorChange(this);
-          }
-          else {
-            this.jsoneditor.onChange();
-          }
-          if (this.progressBar) {
-            this.preview.removeChild(this.progressBar);
-          }
-          uploadButton.removeAttribute("disabled");
-        },
-        failure: (error) => {
-          this.theme.addInputError(this.uploader, error);
-          if (this.progressBar) {
-            this.preview.removeChild(this.progressBar);
-          }
-          uploadButton.removeAttribute("disabled");
-        },
-        updateProgress: (progress) => {
-          if (this.progressBar) {
-            if (progress) {
-              this.theme.updateProgressBar(this.progressBar, progress);
-            }
-            else {
-              this.theme.updateProgressBarUnknown(this.progressBar);
```

Comment:
You're removing this code, but you don't call this function ever anywhere else.  Do you not need this functionality anymore?

Reply:
nope, no one likes progress bars

we prefer progress breweries